django/db/backends/postgresql/features.py

```diff
@@ -26,6 +26,7 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     can_introspect_materialized_views = True
     can_distinct_on_fields = True
     can_rollback_ddl = True
+    schema_editor_uses_clientside_param_binding = True
```

Comment:
Is this in use? I don't see it in this MR. Is it already in the main branch?

I remember introducing [some horrible kludges](https://github.com/dvarrazzo/django/commit/2413142d752bcb4adc1e3696fc6e470d81ad6366) in my original Django psycopg 3 support branch in order to work around the difficulty of generating DML statements using server-side binding: I hope those kludged have been wiped away by a more generous use of client-side binding! :smile:

Reply:
I'll have to double check, maybe it got lost in transition :)

Reply:
Okay, `schema_editor_uses_clientside_param_binding` can probably be removed. As far as your horrible kludges go: @charettes did some impressive work to change our generated queries so that `GROUP BY` (and others) always can use aliases and don't have to either duplicate the whole expression. This nicely removes the need to calculcate the column aliases like you did since there is no more "duplication" of bind params where postgres cannot infer equality.

Reply:
Most of the work Simon did can be found in the commits listed at the top of the page at https://github.com/django/django/commits/main?after=798e38c2b9c46ab72e2ee8c33dc822f01b194b1e+34&branch=main&path%5B%5D=django&path%5B%5D=db&qualified_name=refs%2Fheads%2Fmain

Reply:
Actually `schema_editor_uses_clientside_param_binding` is in use: https://github.com/django/django/blob/5e111c19620d00fec45f365c6d8cc1cc7419a53d/tests/schema/test_logging.py

I also missed grepping for it since it is solely used in the tests. Maybe we can rid of it somehow?

Reply:
> I also missed grepping for it since it is solely used in the tests. Maybe we can rid of it somehow?

I'd leave it.